services/workflows-service/src/providers/file/file-provider/http-file.service.ts

```diff
@@ -1,45 +1,77 @@
+import { HttpService } from '@nestjs/axios';
 import { TLocalFilePath, TRemoteFileConfig, TRemoteUri } from '@/providers/file/types/files-types';
 import { promises as fsPromises } from 'fs';
-import axios, { AxiosResponse } from 'axios';
+import axios, { AxiosError, AxiosInstance, AxiosResponse } from 'axios';
 import { Readable } from 'stream';
 import { IStreamableFileProvider } from '../types/interfaces';
+import { removeSensitiveHeaders } from '@/common/utils/request-response/request';
+import { handleAxiosError } from '@/common/http-service/utils';
+import { LoggerService } from '@nestjs/common';
 
 export class HttpFileService implements IStreamableFileProvider {
-  protected client;
-  constructor(..._args: any) {
-    this.client = axios;
+  protected logger: LoggerService;
+  protected client: AxiosInstance;
+  constructor({ client, logger }: { client: HttpService; logger: LoggerService }) {
+    this.logger = logger;
+    this.client = client.axiosRef;
+
+    this.client.interceptors.request.use(config => {
+      this.logger.debug?.('Axios outgoing request interceptor', {
+        headers: removeSensitiveHeaders(config.headers),
+        url: config.url,
+        method: config.method?.toUpperCase(),
+      });
+
+      return config;
+    });
+
+    this.client.interceptors.response.use(response => {
+      if (
+        Buffer.isBuffer(response.data) ||
+        response.headers['Content-Type'] !== 'application/json'
+      ) {
+        return response;
+      }
+
+      this.logger.debug?.('Axios outgoing response interceptor', {
+        data: response.data,
+        url: response.config.url,
+        method: response.config.method?.toUpperCase(),
+        // TODO: should we add also response's headers?
+      });
+      return response;
+    });
   }
 
+  // private _shouldRetry(error: AxiosError) {
+  //   const { response } = error;
+  //   const isTooManyRequests =
+  //     response?.status === HttpStatusCode.TooManyRequests;
+  //   const isNetworkIssue = error.code === 'ENETUNREACH';
+
+  //   return isTooManyRequests || isNetworkIssue;
+  // }
```

Comment:
Should these comments be removed?

Reply:
because this PR getting to big so i decided to add it as technical debt, this comment is still important